xcresult/src/xcresult.rs

```diff
@@ -0,0 +1,348 @@
+use indexmap::indexmap;
+use lazy_static::lazy_static;
+use quick_junit::{NonSuccessKind, Report, TestCase, TestCaseStatus, TestSuite, XmlString};
+use std::str;
+use std::{fs, process::Command};
+
+#[derive(Debug, Clone)]
+pub struct XCResult {
+    pub path: String,
+    results_obj: serde_json::Value,
+}
+
+const LEGACY_FLAG_MIN_VERSION: i32 = 70;
+
+fn xcrun<T: AsRef<str>>(args: &[T]) -> anyhow::Result<String> {
+    if !cfg!(target_os = "macos") {
+        return Err(anyhow::anyhow!("xcrun is only available on macOS"));
+    }
+    let mut cmd = Command::new("xcrun");
+    let bin = cmd.args(args.iter().map(|arg| arg.as_ref()));
+    let output = bin.output().map_err(|_| {
+        anyhow::anyhow!("failed to run xcrun -- please make sure you have xcode installed")
+    })?;
+    let result = String::from_utf8(output.stdout)
+        .map_err(|_| anyhow::anyhow!("got non UTF-8 data from xcrun output"))?;
+    Ok(result)
+}
+
+fn xcrun_version() -> anyhow::Result<i32> {
+    let version_raw = xcrun(&["--version"])?;
+    // regex to match version where the output looks like xcrun version 70.
+    let re = regex::Regex::new(r"xcrun version (\d+)")?;
+    lazy_static! {
+        static ref RE: regex::Regex = regex::Regex::new(r"xcrun version (\d+)").unwrap();
+    }
+    let res = re
```

Comment:
I think there might have been a small typo here?

```suggestion
    lazy_static! {
        /// regex to match version where the output looks like xcrun version 70.
        static ref RE: regex::Regex = regex::Regex::new(r"xcrun version (\d+)").unwrap();
    }
    let res = RE
```